sweep/sweeper.go

```diff
@@ -1414,17 +1420,16 @@ func (s *UtxoSweeper) attachAvailableRBFInfo(pi *pendingInput) *pendingInput {
 	// - for bitcoind below v24.0.0 we don't have `gettxspendingprevout`.
 	// - for btcd below v0.24.1 we don't have `gettxspendingprevout`.
 	if txOption.IsNone() {
-		return pi
+		return StateInit, fn.None[RBFInfo]()
 	}
 
 	// NOTE: we use UnsafeFromSome for here because we are sure this option
 	// is NOT none.
 	tx := txOption.UnsafeFromSome()
```

Comment:
You *can* use UnsafeFromSome here, but given that you're ultimately returning an Option this might be a situation where `ChainOption` does what you need. However, I'm not sure if we have the other aspects of API infrastructure that will make this possible or worth it.

NOTE: I am not requesting you change this as I can see why you did it. However, I don't like it and I'm trying to figure out what we need to be able to avoid it in the future.

Reply:
Although I do think instead of UnsafeFromSome, you should instead use `UnwrapOrErr` here to prevent panics.

Reply:
yeah i think there maybe something wrong with the design in sweeper, leading up to a case like this. It'd be create to have an API like, `var.Read(func(){when there's some}, func(){when there's none})`, don't know if it's a good idea tho.

Reply:
ok i think i found a way to remove the unsafe usage.

Reply:
ElimOption is that API btw.